x-pack/plugins/lens/public/datasources/form_based/dimension_panel/bucket_nesting_editor.test.tsx

```diff
@@ -99,37 +97,34 @@ describe('BucketNestingEditor', () => {
       />
     );
 
-    component
-      .find('[data-test-subj="indexPattern-nesting-switch"] button')
-      .first()
-      .simulate('click');
-
+    fireEvent.click(screen.getByTestId('indexPattern-nesting-switch'));
```

Comment:
Nice improvement! 
Would it be possible to make the whole test that uses events `async` and then use 
```suggestion
    await userEvent.click(screen.getByTestId('indexPattern-nesting-switch'));
```
(same for the other tests)

Reply:
They are updated and pass much quicker now:
<img width="1102" alt="Screenshot 2024-11-15 at 3 23 56 PM" src="https://github.com/user-attachments/assets/d00d803b-a6aa-4156-9c91-03266054849b">
Thanks! 😊

Reply:
Amazing! I'll go ahead and approve it now, thank you for all the hard work 🥇